.github/workflows/ci.yml

```diff
@@ -102,6 +102,7 @@ jobs:
         ctest -T memcheck --output-on-failure
 
     - uses: actions/upload-artifact@v2
+      if: failure()
```

Comment:
see https://stackoverflow.com/a/58859404

now it will upload the artifacts only if it failed